main.py

```diff
@@ -118,128 +124,236 @@ async def wrapper(*args, **kwargs):
+                        landfield_details = await call_api(landfield_details_url)  # Use a helper function to make API calls
+
+                        # Access promisedEssenceBalance from landfield details
+                        promised_essence = Decimal(landfield_details["promisedEssenceBalance"])
+                        claimed_essence = Decimal(landfield_details["claimedEssenceBalance"])
+
+                        processing_str = "\t<< Processing A.T. " + str(at_landfield_id) + ": "
+                        processing_str += "[ TIER: "+str(at_tile_tier)
+                        processing_str += " | TILES: "+str(at_tile_count) 
+                        processing_str += " | CIVS: " + str(at_civs) 
+                        processing_str += " | DROIDS: " + str(at_droids) 
+                        processing_str += " | FAVORITE: " + str(at_favorited) + "]"
+                        print( processing_str )
+                        # break out of loop, we raided recently (assumed mihaj script)
+                        if( at_is_raided ):
+                            print("\t\t<< Recently raided >>")
+                            continue
+                        # break out of loop, we reached our breaker
+                        if( at_tile_count < TILE_BREAKER_COUNT ):
+                            print("\t\t<< Breaker threshold reached >>")
+                            return
+                        if (
+                            at_droids == 0
+                            and at_civs == 0
+                            and at_favorited is None
+                            and (promised_essence > 0 or claimed_essence < 1)
+                            and at_is_raided is False
+                        ):
+                            favorite_payload = await call_favorite_bulk(at_landfield_id)
+                            print("\t\t" + "|| Favorite Added || " + str(favorite_payload))
+                            new_favorites_added = True
+                            await wait_helper(level=2)
+                            
+                    except Exception as e:
+                        print(f"\t\tError processing target {target['id']}: {e}")
+                        continue  # Skip to the next target
+
+            if (i + 1) < total_at_pages:
+                print(f"\t++ A.T. Page Flip to Page {i + 1} ++")
+                payload = await call_get_attack_targets(property_id, i + 1, droids_url_str)
+                await wait_helper(level=1)
+                
+    except Exception as e:
+        print(f"Error fetching attack targets for property {property_id}: {e}")
+
+    if new_favorites_added == True:
+        with open("new_favorites.txt", "a") as f:
+            f.write(f"{property_id} - {property_description}\n")
+
+def input_thread():
+    global skip_to_next
+    while True:
+        if msvcrt.kbhit() and msvcrt.getch() == b'y':
```

Comment:
should probably refactor this portion.  msvcrt ties us to MS and will break (afaik) on linux.